docs/2-the-zarf-cli/2-declarative-cli-configuration.md

```diff
@@ -0,0 +1,48 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ExampleYAML from '@site/src/components/ExampleYAML';
+import FetchFileCodeBlock from '@site/src/components/FetchFileCodeBlock';
+
+# Declarative CLI Configuration
+
+## Overview
+
+Users can configure the `zarf init`, `zarf package create`, and `zarf package deploy` command flags, as well as global flags (with the exception of `--confirm`), through a config file to help execute commands more declaratively.
+
+By default, Zarf searches for a config file named `zarf-config.toml` in the current working directory. You can generate a config template for use by Zarf by executing the command `zarf prepare generate-config`, with an optional filename, in any of the supported formats, including `toml`, `json`, `yaml`, `ini` and `props`. For instance, to create a template config file with the `my-cool-env` in the yaml format, you can use the command `zarf prepare generate-config my-cool-env.yaml`.
```

Comment:
```suggestion
Zarf supports config files written in common configuration file formats including `toml`, `json`, `yaml`, `ini` and `props`, and by default Zarf will look for a file called `zarf-config` with one of these filenames in the current working directory.  To generate a blank config file you can run `zarf prepare generate-config` with an optional output filename/format.  For example, to create an empty config file with the `my-cool-env` in the yaml format, you can use `zarf prepare generate-config my-cool-env.yaml`.
```

- by Zarf by is awkward wording
- executing the command `zarf ...` is over-explanatory
- filenames might be best explained first since Zarf will always look for all of them